.github/workflows/reusable-docker-build-publish.yml

```diff
@@ -255,10 +251,20 @@ jobs:
           GITHUB_EVENT_NAME: ${{ inputs.github-event-name }}
           GITHUB_REF_NAME: ${{ inputs.github-ref-name }}
         run: |
-          echo "Setting extra outputs..."
-          if [[ "${GITHUB_EVENT_NAME:-}" == "pull_request" ]]; then
+          echo "Setting build type outputs..."
+          if [[ "${GITHUB_EVENT_NAME}" == "schedule" ]]; then
+            echo "build-type=nightly" | tee -a "$GITHUB_OUTPUT"
+          elif [[ "${GITHUB_EVENT_NAME}" == 'push' ]]; then
+            echo "build-type=branch" | tee -a "$GITHUB_OUTPUT"
+          elif [[ "${GITHUB_EVENT_NAME}" == 'pull_request' ]]; then
             pr_number=$(echo "${GITHUB_REF_NAME}" | cut -d'/' -f1)
             echo "pr-number=${pr_number}" | tee -a "${GITHUB_OUTPUT}"
+            echo "build-type=pr" | tee -a "$GITHUB_OUTPUT"
+          elif [[ "${GITHUB_EVENT_NAME}" == 'workflow_dispatch' ]]; then
+            echo "build-type=manual" | tee -a "$GITHUB_OUTPUT"
+          else
+            echo "::error::Unsupported GitHub event name: ${GITHUB_EVENT_NAME}"
```

Comment:
How do we support a tag trigger here, or does an override needs to be used for that? How would conditionally setting the override look like?

Reply:
Added support in 310d49c. You'll have to pass in a new input from your workflow:

```
...
with:
  ...
  github-ref-type: ${{ github.ref_type }}
  ...
```

`github.ref_type` will either be `tag` or `branch`.